src/views/Verifier.js

```diff
@@ -1,204 +1,142 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import QrReader from "react-qr-reader";
 import QRCode from "react-qr-code";
-// import { useMoralis, useMoralisWeb3Api } from "react-moralis";
-import { FcCancel, FcTouchscreenSmartphone } from "react-icons/fc";
+import { FcCancel } from "react-icons/fc";
 import { Checkmark } from "react-checkmark";
-import { padding } from "@mui/system";
-import axios from "axios";
-
-class Verifier extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      attendee_addr: "",
-      creator_addr: "",
-      dataCollectionStep: 0,
-      mintDate: "",
-      isValidAttendee: false,
-    };
+import { useMoralis } from "react-moralis";
+import moment from "moment";
+import { Audio } from "react-loader-spinner";
+
+function Verifier() {
+  const [attendee_address, set_attendee_address] = useState("");
+  const [creator_address, set_creator_address] = useState("");
+  const [mint_date, set_mint_date] = useState("");
+  const [is_valid_attendee, set_is_valid_attendee] = useState(0);
+  const { Moralis } = useMoralis();
+
+  function moralisDateToUnixTimestamp(date_string) {
+    return moment(date_string.split(".")[0], "YYYY-MM-DDTHH:mm:ss").unix();
   }
 
-  async verifyAttendee() {
-    const nft_port_api_key = "cb0130f1-d009-4f9d-9f5b-f9c6471ebf38";
-    const moralis_api_key =
-      "gxpDcsiOxjhx1SvF5ROBiMVeeGsQ9WtZSo9mDB1nWpNf9hnd4aOpzU91A2jABQi0";
-    // const { initialize, isInitialized } = useMoralis();
-    // initialize({
-    //   appId: "ICYNGMsC7ru5DREZVwoWjX39UcYQo34HAsAIOQHo",
-    //   serverUrl: "https://hxzbhq0gqxph.usemoralis.com:2053/server",
-    // });
-    const nftport_instance = axios.create({
-      baseURL: "https://api.nftport.xyz/v0/",
-      headers: {
-        Authorization: nft_port_api_key,
-        "Content-Type": "application/json",
-      },
-    });
-
-    const response = await nftport_instance.get(
-      `accounts/${this.state.creator_addr}?chain=ethereum`
+  async function verifyAttendee() {
+    const moralis_options = {
+      chain: "mumbai",
+      address: creator_address,
+      limit: "500",
+      direction: "to",
+    };
+
+    const transfers_to_creator = await Moralis.Web3API.account.getNFTTransfers(
+      moralis_options
     );
-    if (!response.ok) {
-      console.log("Network response not OK!");
+    console.log("fetched transactions to " + creator_address);
+
+    const nfts_from_attendee = transfers_to_creator.result
+      .filter((transaction) => transaction.from_address === attendee_address)
+      .map((transaction) => transaction.token_address + transaction.token_id);
```

Comment:
if we don't put a space, comma or something between these or store them as tuples, we might have some issues decoupling them.

(ofc we do know the exact length of an address but that would be kinda messy)

Reply:
nm this is just for a boolean value. could still be nice to add something between for debugging purposes down the line